kedro/io/core.py

```diff
@@ -871,3 +881,70 @@ def validate_on_forbidden_chars(**kwargs: Any) -> None:
             raise DatasetError(
                 f"Neither white-space nor semicolon are allowed in '{key}'."
             )
+
+
+_C = TypeVar("_C")
```

Comment:
What is this for?

Reply:
This type parameter for the catalog which is needed to define a generic protocol later in the code since `from_config()` and `shallow_copy()` return catalog type.

https://peps.python.org/pep-0544/#generic-protocols